ArdentHQ/dashbrd
# refactor: store erc1155 tokens

## Summary

https://app.clickup.com/t/862kfgzfk + https://app.clickup.com/t/8678v862p

This PR adds a `type` column to the table `collections` table, which indicates the type of a collection (ERC1155 or ERC721). Previously, we had been filtering out non-ERC721 tokens, but since we want to support ERC1155 in the future + show the message to the user that they own ERC1155 tokens, we'll no longer filter them out when returned by the provider and store them. This PR will also show the little "warning" toast if the user owns any ERC1155 tokens.

However, we won't display those ERC1155 collections anywhere and ignore them in jobs and scheduled commands. And if the collection is ERC1155, we won't index its NFTs.

We previously already filtered out non-ERC721, so only the seeded collections can potentially be ERC1155, and as it turns out, there's a lot of them in our dump data (https://github.com/ArdentHQ/dashbrd-data/blob/main/live-dump/collection-nfts/top-eth-collections.json and https://github.com/ArdentHQ/dashbrd-data/blob/main/live-dump/collection-nfts/top-polygon-collections.json, look for `erc_type` column). So this PR will hardcode these existing addresses as they do exist in prod and staging, and set their type to ERC1155.

To determine whether a wallet owns any ERC1155 (to show in the UI), we take their every NFT from the provider and check for ERC1155 type before filtering them out, and update the boolean flag for the user. If it does, the wallet does own it. As we schedule the job to update their NFTs, this will be updated with time. This needs to be done in this way because we don't store ERC1155 NFTs, only the collections -- but the `nfts` table is the link between the wallet and the collection.

### Testing ERC1155 toast

1. Connect the wallet that owns ERC1155 and let it index everything for the wallet
2. Open up Dashbrd and see the toast
3. Click X to hide it
4. Won't show again unless you clear `localStorage`
5. Notice how there are some collections in the database that have ERC1155 type, but they don't have any NFTs indexed

### Testing type

1. `git checkout develop`
2. import the collections dump for prod or staging from Google Drive
3. `git checkout refactor/store-erc1155`
4. `php artisan migrate`
6. Open the database GUI and see how types are set even for existing collections







## Checklist
